keps/sig-auth/20200507-external-tls-certificate-authenticator/README.md

```diff
@@ -0,0 +1,749 @@
+-->
+
+This enhancement includes unit and integration tests:
+
+- Unit tests in `pkg/client/auth/externalsigner` to:
+  - test that the external singer authentication provider APIs follow the format
+    defined in the specification,
+  - test the internal mechanisms of the authentication provider, such as
+    caching, and
+  - test handling of certificates and signatures data (including (un)marshalling
+    the messages and en/decoding values).
+
+- Integration tests in `test/integration/auth/externalsigner_test.go` to:
+  - attempt an execution of a `kubectl` command with authentication using the
+    external singer authentication provider.
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, or as something else. The KEP
+should keep this high-level with a focus on what signals will be looked at to
+determine graduation.
+
+Consider the following in developing the graduation criteria for this
+enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning),
+or by redefining what graduation means.
+
+In general, we try to use the same stages (alpha, beta, GA), regardless how the
+functionality is accessed.
+
+[maturity-levels]:
+https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]:
+https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity
+levels][maturity-levels].
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and surveys
+- Complete features A, B, C
+- Tests are in Testgrid and linked in KEP
+
+#### Beta -> GA Graduation
+
+- N examples of real world usage
+- N installs
+- More rigorous forms of testing e.g., downgrade tests and scalability tests
+- Allowing time for feedback
+
+**Note:** Generally we also wait at least 2 releases between beta and GA/stable,
+since there's no opportunity for user feedback, or even bug reports, in
+back-to-back releases.
+
+#### Removing a deprecated flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality which deprecates the
+  flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]:
+https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+-->
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback regarding the API from developers of external plugins
```

Comment:
might be worth outlining the plans for the future for -> Beta and -> GA graduation too.

Reply:
+1